dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MinimumJavaVersionChecker.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MinimumJavaVersionChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(MinimumJavaVersionChecker.class);
+
+  static void check(String version) {
+    if (!version.startsWith("1.")) {
+      return;
+    }
+    Pattern oldVersionPattern = Pattern.compile("1\\.([0-9]+)");
```

Comment:
This is a non-anchored pattern used with find(), so it will match things like 21.7 and return 1.7

Reply:
It will not match `21.7`, because there is `version.startsWith("1.")` check in the `if` above.

Reply:
I've added the anchor.